src/assets/wise5/vle/node/nodeController.ts

```diff
@@ -879,6 +879,22 @@ class NodeController {
    * @param componentState the new component state
    */
   notifyConnectedParts(changedComponentId, componentState) {
+    for (const subscribingComponent of this.getComponents()) {
+      if (subscribingComponent.connectedComponents != null) {
+        for (const publishingComponent of subscribingComponent.connectedComponents) {
+          if (
+            publishingComponent.nodeId === componentState.nodeId &&
+            publishingComponent.componentId === componentState.componentId
+          ) {
+            this.ComponentService.notifyConnectedComponents(
+              this.nodeId,
+              subscribingComponent.id,
+              componentState
+            );
+          }
+        }
+      }
```

Comment:
This code block is a bit confusing, maybe because of the variable names subscribingComponent and publishingComponent. From my understanding, not all components in this.getComponents() are subscribingComponents, so should it be replaced with component? Can publishingComponent be replaced with connectedComponent? 

Also, we don't check for changedComponentId != null and componentState != null in this block like we do on like 898. Can we assume that those passed-in params will be non-null and remove the null checks?

Reply:
I've changed the variable names. I also changed the notifyConnectedComponents function name because that's what I was originally trying to make clearer. We're not notifying the connected components. We're actually notifying the component that subscribes to the connected component.

The code at 898 and below will be removed once all the components subscribe to the observable so we don't need to worry about that. The changedComponentId parameter probably won't be needed in the future either so we'll get rid of that later. We can get the nodeId and componentId from the componentState.